R/all_endemic_clades.R

```diff
@@ -0,0 +1,42 @@
+#' Checks if all island species are endemic clades. Clade are colonisations that
+#' have formed an island clade of two or more species surviving to the present.
+#'
+#' @inheritParams default_params_doc
+#'
+#' @return Boolean
+#' @author Joshua W. Lambert
+#' @export
+all_endemic_clades <- function(sim) {
+  testit::assert(is.list(sim))
+  all_endemic <- all_endemic(sim)
+  if (isFALSE(all_endemic)) {
+    return(FALSE)
+  }
+  sim <- sim[-1]
+  branching_times <- lapply(sim, "[[", "branching_times")
+  num_spec <- unlist(lapply(branching_times, function(x) {
+    length(x) - 1
+  }))
+  stacs <- unlist(lapply(sim, "[[", "stac"))
+  clades <- all(num_spec[which(stacs != 3)] > 1)
+  if (any(stacs == 3)) {
```

Comment:
This can be done by calling `any_recols()` unless I'm missing something, no? I'm happy to change if you agree